src/Illuminate/Routing/Router.php

```diff
@@ -719,7 +724,7 @@ public function resolveMiddlewareClassName($name)
 
         list($name, $parameters) = array_pad(explode(':', $name, 2), 2, null);
 
-        return array_get($map, $name, $name).($parameters ? ':'.$parameters : '');
+        return (isset($map[$name]) ? $map[$name] : $name).($parameters ? ':'.$parameters : '');
```

Comment:
Avoid using `array_get()` on middleware name to allow using '.' as route middleware name.